segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/metadata/StorageMetadataException.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.metadata;
+
+/**
+ * Exception related to storage metadata operations.
+ */
+public class StorageMetadataException extends Exception {
```

Comment:
What pattern have we been using for exceptions? I'm wondering whether we should group the subtypes in the same class definition like in `KeeperException`:

https://zookeeper.apache.org/doc/r3.6.0/apidocs/zookeeper-server/index.html?org/apache/zookeeper/KeeperException.html

I don't necessarily want to deviate from how we have been doing it elsewhere in the code.

Reply:
We are following current convention in the code.